src/crypto/ms_oapxbc.rs

```diff
@@ -29,72 +24,56 @@ const CTX_NONCE_LEN: usize = 32;
 pub enum MsOapxbcSessionKey {
     /// An AES-256-GCM/CBC + HS256 session key
     A256GCM {
-        /// The aes key for this session
-        aes_key: [u8; A256_KEY_LEN],
+        /// The encrypted session key which can be loaded as required for individual operations
+        loadable_session_key: LoadableMsOapxbcSessionKey,
     },
 }
 
-#[cfg(test)]
-impl MsOapxbcSessionKey {
-    pub(crate) fn assert_key(&self, key: &[u8]) -> bool {
-        match self {
-            MsOapxbcSessionKey::A256GCM { aes_key, .. } => aes_key == key,
-        }
-    }
-}
-
 impl MsOapxbcSessionKey {
-    /// Given a public key, create a derived session key. This is the "server side"
-    /// component for this process.
-    pub fn begin_rsa_oaep_key_agreement(
-        rsa_pub_key: Rsa<Public>,
-    ) -> Result<(Self, JweCompact), JwtError> {
-        let rsa_oaep = JweRSAOAEPEncipher::try_from(rsa_pub_key)?;
-
-        let aes_key = JweA256GCMEncipher::new_ephemeral()?;
-
-        // https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-oapxbc/d3f8893c-a715-412a-97d1-5ffed2c3b3d5
-        // session_key_jwe does not seem to have many requirements?
-        let jweb = JweBuilder::from(Vec::with_capacity(0)).build();
-        let jwec = aes_key.encipher_inner(&rsa_oaep, &jweb)?;
-
-        // Extract the raw key.
-        let aes_key = aes_key.raw_key();
-
-        Ok((MsOapxbcSessionKey::A256GCM { aes_key }, jwec))
-    }
-
     /// Given a session jwe, complete the session key derivation with this private key
-    pub fn complete_rsa_oaep_key_agreement(
-        rsa_priv_key: Rsa<Private>,
+    pub fn complete_tpm_rsa_oaep_key_agreement<T>(
```

Comment:
I'd like to see complete_tpm_rsa_oaep_key_agreement part of the hsm-crypto feature, and complete_rsa_oaep_key_agreement still available without hsm-crypto.